crates/core/src/execution_context.rs

```diff
@@ -37,15 +43,17 @@ impl<'a> ExecutionContext<'a> {
         Self {
             database,
             reducer: Some(name),
+            query_debug_info: None,
             txn_type: TransactionType::Reducer,
         }
     }
 
     /// Returns an [ExecutionContext] for a sql or subscription transaction.
-    pub fn sql(database: Address) -> Self {
+    pub fn sql(database: Address, query_debug_info: QueryDebugInfo) -> Self {
```

Comment:
I guess it could be. QueryDebugInfo wanted to be an Arc because it gets stuck in places where it gets cloned, so it's a little less restrictive to not make it one, but I can make it one for uniformity.